{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f93e9cf7-6d9d-4244-8854-68d99be4aaa6",
   "metadata": {},
   "source": [
    "# The purpose of this notebook is to condense and simplify the running of FoldSeek, and useful downstream processes into a single Jupyter Notebook. The objective is that a user can define several parameters and directories and then click 'Run all' and leave the notebook to finish for however long it takes.\n",
    "\n",
    "# This is meant to serve as an example so PLEASE DO NOT MODIFY, to use this notebook, please setup a new directory and copy in this notebook (you'll have to change some direcotry names for this), or run the accompanying script via the command line - this will take as inputs the paths to various directories it needs via the command line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "d6e2c70a-4ae4-49aa-acb0-02fe9dc5bbbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import time\n",
    "import glob\n",
    "%run functions.py"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a70713e-da92-4fca-bae5-4b4ca4c1a077",
   "metadata": {},
   "source": [
    "### A few things to note:\n",
    "\n",
    "1. Camel Case is used AT ALL TIMES for variables, Snake case is used for functions - this way if there are any name clashes functions and variables can be distinguished by the user"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "711af46c-ad30-497c-b682-d1bfaa6040cf",
   "metadata": {},
   "source": [
    "## Parameters"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93942e64-6092-4c36-b2c0-b8aae892af60",
   "metadata": {},
   "source": [
    "### Parameters and directories for FoldSeek\n",
    "You need to create the search directory where everything will go and fill it witha  sub-directory of query proteins, everything else should be done for you. Check everything makes sense at this stage!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "a4a1381e-6c8c-42f4-9143-4a4807a508fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['FoldSeekExamples/ProteinQueries/1rei.pdb',\n",
       " 'FoldSeekExamples/ProteinQueries/1mbn.pdb']"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CurrentSearchDirectory = 'FoldSeekExamples' # The directory which houses and will house all sub-directories e.g. CancerCureFoldSeekSearch\n",
    "QueryPath = f'{CurrentSearchDirectory}/ProteinQueries'\n",
    "assert os.path.exists(QueryPath), f\"Directory does not exist: {QueryPath}\" # Assert that the folder with protein queries already exists\n",
    "QueryProteins = list_files_in_directory(QueryPath)\n",
    "assert (len(QueryProteins)>0), f'No proteins in directory: {QueryPath}'\n",
    "QueryProteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "9cd2bf78-63d7-4955-82d4-f4033660bcfc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['databases/UP/up',\n",
       " 'databases/ESMA/esma',\n",
       " 'databases/AFDB/afdb',\n",
       " 'databases/PDB/pdb',\n",
       " 'databases/SP/sp']"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DatabasePath = 'databases/' # The databases that FoldSeek will look through are in this directory - there are 5\n",
    "Databases = list_folders_in_directory(DatabasePath)\n",
    "Databases = [f\"{element.rsplit('/', 1)[0]}/{element.rsplit('/', 1)[1]}/{element.rsplit('/', 1)[1].lower()}\" if '/' in element else element for element in Databases]\n",
    "assert (len(Databases)==5), f'Five databases should be present in the database directory!'\n",
    "Databases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f5b64b2d-3704-446d-b5e8-a723df67fe6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# The below variables will tell FoldSeek what to output, and the corresponding column names they will be stored in in the final CSV\n",
    "# I highly reccomend looking at the Github for FoldSeek and MMSeq2 which detail what these variables are adn how they're computed\n",
    "# Once this is studied, you will notice tCa and qCa (co-ordinates of query and target alpha carbons) are missing - they were \n",
    "# creating issues with the tab-separated readout, this may be corrected in future updates so check if this is still the case if you think\n",
    "# these variables will be useful to you.\n",
    "\n",
    "FormatOutput = (\"--format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,\"\n",
    "                \"alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\") \n",
    "\n",
    "Columns = ['query','target','fident','alnlen','mismatch','gapopen','qstart','qend','tstart','tend','tseq','evalue','bits','cigar',\n",
    "           'alntmscore','qtmscore','ttmscore','rmsd','u','t','lddt','lddtfull','prob','taxid','taxname','taxlineage','database']\n",
    "\n",
    "# The ESMA database has no taxonomic data, and so requires its own handling\n",
    "FormatOutputESMA = (\"--format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,\"\n",
    "                \"alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob\") \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a5228b42-0421-4f2d-9aff-c4bccb88a694",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directory 'FoldSeekExamples/FoldSeekExtractionOutputCSVs' was created.\n",
      "Directory 'FoldSeekExamples/FoldSeekExtractionOutputHTMLs' was created.\n",
      "Directory 'FoldSeekExamples/FoldSeekExtractionOutputAlntmps' was created.\n"
     ]
    }
   ],
   "source": [
    "CSVFileDirectory = f\"{CurrentSearchDirectory}/FoldSeekExtractionOutputCSVs\" # Several CSV files will be generated and placed here\n",
    "check_existence(CSVFileDirectory)\n",
    "CreateHTMLs = True # Will we generate HTMLs? - expect a doubling of runtime if you do\n",
    "if CreateHTMLs:\n",
    "    HTMLFileDirectory = f\"{CurrentSearchDirectory}/FoldSeekExtractionOutputHTMLs\" # If we are making HTMLs we'll put them here\n",
    "    check_existence(HTMLFileDirectory)\n",
    "AlignmentAndTempFileDirectory = f\"{CurrentSearchDirectory}/FoldSeekExtractionOutputAlntmps\"\n",
    "check_existence(AlignmentAndTempFileDirectory)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4dbd9410-d34f-4c25-b069-62a3604853de",
   "metadata": {},
   "source": [
    "### Parameters and directories for BLASTs\n",
    "These parameters are almost entirely user-dependent so all of this is up to you! Put your databases in the BLASTDatabases folder, reading documentation on the BLAST functions will probably be very useful. Something to bear in mind is that you can create a BLASTable database from a fasta file with all the sequences you want, or download a database with all the needed files ready-made from NCBI etc. Examples of both are shown here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "c0e64e5e-b88c-4d64-b280-ae40e4bc352e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directory 'FoldSeekExamples/BLAST_XMLs' was created.\n"
     ]
    }
   ],
   "source": [
    "BLASTOutputDirectory = f'{CurrentSearchDirectory}/BLAST_XMLs' # BLASTs will be saved and parsed as .xml files, please read documentation if not familiar\n",
    "check_existence(BLASTOutputDirectory)\n",
    "BLASTList = ['BPPRC','AllergenOnline','Patent']# The list of BLASTs that we will be doing, in the order we will do them\n",
    "BLASTDirectories = [f'BLASTDatabases/{BLAST}_BLAST_DB' for BLAST in BLASTList]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "34bd2833-9540-4430-8cf3-d63ffbf18215",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directory 'BLASTDatabases/BPPRC_BLAST_DB' already exists.\n",
      "Directory 'BLASTDatabases/AllergenOnline_BLAST_DB' already exists.\n",
      "Directory 'BLASTDatabases/Patent_BLAST_DB' already exists.\n"
     ]
    }
   ],
   "source": [
    "for directory in BLASTDirectories: # make sure they exist!\n",
    "    check_and_assert_existence(directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "757301e1-ce23-400e-9c29-494f5dc59037",
   "metadata": {},
   "outputs": [],
   "source": [
    "RebuildDBs = True # If this is true, an if statement will be entered where the BLASTable database is built up from a single fasta file\n",
    "DBsToRebuild = ['BPPRC','AllergenOnline'] # What DBs need to be rebuilt in such a way - in the example these two, \n",
    "                                          #the patent is from NCBI so will already contain everything necessary (and doesn't have a fasta!)\n",
    "if RebuildDBs:\n",
    "    for DB in DBsToRebuild:\n",
    "        PathToDB = f'BLASTDatabases/{DB}_BLAST_DB'\n",
    "        DBFASTA = glob.glob(f'{PathToDB}/*.fa')[0] # finds the fasta in the directory - obviously make sure its there!!!\n",
    "        build_blast_database(DBFASTA, dbtype='prot') # Builds out the necessary files for BLASTing"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "31a0d0dc-99c6-45b9-999e-557f1a758293",
   "metadata": {},
   "source": [
    "### Parameters for filtering based on structural (FoldSeek) or sequential (BLAST) alignments\n",
    "The world is your oyster, here are some examples that will probably hold true for most runs, but you could add a filter based on if the taxname is Equus unicornus, or if a certain subsequence appears in the target sequence, or if the BLAST description says 'Cures all cancers'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "224f65b7-1c28-4781-a1b2-8a227a95fb0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "TMScoreThreshold = 0.4 # NOTE: this will be applied to ttmscore AND qtmscore, if you don't know what these are, please read the documentation!\n",
    "BLASTBitscoreThresholds = [100,100,50] # Only those sequences whose *BEST* BLAST is a worse match than these bitscores go through\n",
    "                                       # In this example, 100 for BPPRC,100 for AllergenOnline, 50 for Patent\n",
    "                                       # To get a feel of bitscore read documentation, and visualise yourself!\n",
    "MustStartWithMethionine=True    # If you want to clone it, it needs to have a start codon...\n",
    "RemoveDuplicates = True # Get rid of duplicate proteins so we have an idea of what we really want to keep"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c487547-1337-4adb-9db4-2e7d44f8da4c",
   "metadata": {},
   "source": [
    "# Code\n",
    "## Obtaining variables from FoldSeek\n",
    "### I reccomend looking at the FoldSeek Github repo to understand the command line commands being built here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9b75cdb6-9654-4909-91ff-ef96a87f81df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_up.html exists and will be overwritten\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_up.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query\n",
      "[=================================================================] 1 0s 12ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 1ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 39ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query_ss databases/UP/up_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp/16199660528816721581/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 276G\n",
      "Target database size: 53665860 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 53.67M 18s 693ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 53.67M 1m 1s 230ms\n",
      "Index statistics\n",
      "Entries:          11018018466\n",
      "DB size:          63533 MB\n",
      "Avg k-mer size:   172.156539\n",
      "Top 10 k-mers\n",
      "    DDDDDD\t25475954\n",
      "    DDDDDP\t22448571\n",
      "    DDDDPP\t19428803\n",
      "    VVVVPD\t17790099\n",
      "    DPVVVV\t16769600\n",
      "    SVSVVV\t14134938\n",
      "    SVVVVV\t13707739\n",
      "    DDVVVV\t12394056\n",
      "    LVVVVV\t11880759\n",
      "    VSVVVV\t10654575\n",
      "Time for index table init: 0h 3m 0s 648ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 53665860\n",
      "[=================================================================] 2 0s 2ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "860224836 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 3m 21s 882ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp/16199660528816721581/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp/16199660528816721581/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 684ms\n",
      "Time for merging to strualn: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 1m 14s 405ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp/16199660528816721581/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp/16199660528816721581/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp/16199660528816721581/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/search_tmp/16199660528816721581/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_up.html exists and will be overwritten\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_up.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 74ms\n",
      "Time for merging to 1rei_up.html: 0h 0m 0s 11ms\n",
      "Time for processing: 0h 0m 17s 310ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/8403727329891312885/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_up exists and will be overwritten\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query\n",
      "[=================================================================] 1 0s 13ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query_ss databases/UP/up_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp/12898501773263263249/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 276G\n",
      "Target database size: 53665860 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 53.67M 18s 481ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 53.67M 1m 2s 626ms\n",
      "Index statistics\n",
      "Entries:          11018018466\n",
      "DB size:          63533 MB\n",
      "Avg k-mer size:   172.156539\n",
      "Top 10 k-mers\n",
      "    DDDDDD\t25475954\n",
      "    DDDDDP\t22448571\n",
      "    DDDDPP\t19428803\n",
      "    VVVVPD\t17790099\n",
      "    DPVVVV\t16769600\n",
      "    SVSVVV\t14134938\n",
      "    SVVVVV\t13707739\n",
      "    DDVVVV\t12394056\n",
      "    LVVVVV\t11880759\n",
      "    VSVVVV\t10654575\n",
      "Time for index table init: 0h 2m 59s 850ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 53665860\n",
      "[=================================================================] 2 0s 2ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "860224836 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 3m 20s 784ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp/12898501773263263249/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp/12898501773263263249/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 770ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 1m 15s 426ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp/12898501773263263249/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp/12898501773263263249/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp/12898501773263263249/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/search_tmp/12898501773263263249/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_up exists and will be overwritten\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_up --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 807ms\n",
      "Time for merging to aln_1rei_up: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 18s 236ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10845808999738407823/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_esma.html exists and will be overwritten\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_esma.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query\n",
      "[=================================================================] 1 0s 14ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 42ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query_ss databases/ESMA/esma_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp/13838365028824419457/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 175G\n",
      "Target database size: 36986627 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 36.99M 10s 490ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 36.99M 33s 267ms\n",
      "Index statistics\n",
      "Entries:          6585984425\n",
      "DB size:          38173 MB\n",
      "Avg k-mer size:   102.906007\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t11204510\n",
      "    VVLVVV\t9981536\n",
      "    SVSVVV\t9579775\n",
      "    VVSVVV\t7745515\n",
      "    SVVVVV\t7513922\n",
      "    VSVVVV\t5273955\n",
      "    VVNVVV\t4240901\n",
      "    DPSVVV\t4218901\n",
      "    DDVVVV\t4026947\n",
      "    PPVVVV\t3260480\n",
      "Time for index table init: 0h 1m 33s 3ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 36986627\n",
      "[=================================================================] 2 0s 2ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "612866034 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 1m 45s 516ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp/13838365028824419457/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp/13838365028824419457/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 57ms\n",
      "Time for merging to strualn: 0h 0m 0s 6ms\n",
      "Time for processing: 0h 1m 5s 267ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp/13838365028824419457/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp/13838365028824419457/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp/13838365028824419457/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/search_tmp/13838365028824419457/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_esma.html exists and will be overwritten\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_esma.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 131ms\n",
      "Time for merging to 1rei_esma.html: 0h 0m 0s 11ms\n",
      "Time for processing: 0h 0m 5s 714ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/5231641953223275196/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_esma exists and will be overwritten\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query\n",
      "[=================================================================] 1 0s 7ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query_ss databases/ESMA/esma_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp/6173723353360577952/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 175G\n",
      "Target database size: 36986627 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 36.99M 9s 158ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 36.99M 31s 744ms\n",
      "Index statistics\n",
      "Entries:          6585984425\n",
      "DB size:          38173 MB\n",
      "Avg k-mer size:   102.906007\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t11204510\n",
      "    VVLVVV\t9981536\n",
      "    SVSVVV\t9579775\n",
      "    VVSVVV\t7745515\n",
      "    SVVVVV\t7513922\n",
      "    VSVVVV\t5273955\n",
      "    VVNVVV\t4240901\n",
      "    DPSVVV\t4218901\n",
      "    DDVVVV\t4026947\n",
      "    PPVVVV\t3260480\n",
      "Time for index table init: 0h 1m 31s 447ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 36986627\n",
      "[=================================================================] 2 0s 2ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "612866034 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 1m 43s 917ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp/6173723353360577952/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp/6173723353360577952/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 710ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 39s 664ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp/6173723353360577952/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp/6173723353360577952/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp/6173723353360577952/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/search_tmp/6173723353360577952/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_esma exists and will be overwritten\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_esma --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 364ms\n",
      "Time for merging to aln_1rei_esma: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 5s 668ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11987099034259865940/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_afdb.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query\n",
      "[=================================================================] 1 0s 7ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 1ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 42ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query_ss databases/AFDB/afdb_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp/6688187299135845983/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 4G\n",
      "Target database size: 564446 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 564.45K 0s 313ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 564.45K 0s 856ms\n",
      "Index statistics\n",
      "Entries:          178833320\n",
      "DB size:          1511 MB\n",
      "Avg k-mer size:   2.794271\n",
      "Top 10 k-mers\n",
      "    DDDDDD\t343438\n",
      "    DDDDDP\t320181\n",
      "    DDDDPP\t287311\n",
      "    VVVVPD\t239933\n",
      "    VVVVPP\t215141\n",
      "    LVLVVV\t207981\n",
      "    DPVVVV\t204183\n",
      "    SVSVVV\t197597\n",
      "    SVVVVV\t182723\n",
      "    LVVVVV\t158216\n",
      "Time for index table init: 0h 0m 2s 801ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 564446\n",
      "[=================================================================] 2 0s 2ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "11833732 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 312ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp/6688187299135845983/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp/6688187299135845983/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 207ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 1s 865ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp/6688187299135845983/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp/6688187299135845983/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp/6688187299135845983/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/search_tmp/6688187299135845983/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_afdb.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 247ms\n",
      "Time for merging to 1rei_afdb.html: 0h 0m 0s 18ms\n",
      "Time for processing: 0h 0m 0s 777ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/714613885492735869/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query\n",
      "[=================================================================] 1 0s 13ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query_ss databases/AFDB/afdb_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp/5123411441862688475/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 4G\n",
      "Target database size: 564446 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 564.45K 0s 319ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 564.45K 0s 823ms\n",
      "Index statistics\n",
      "Entries:          178833320\n",
      "DB size:          1511 MB\n",
      "Avg k-mer size:   2.794271\n",
      "Top 10 k-mers\n",
      "    DDDDDD\t343438\n",
      "    DDDDDP\t320181\n",
      "    DDDDPP\t287311\n",
      "    VVVVPD\t239933\n",
      "    VVVVPP\t215141\n",
      "    LVLVVV\t207981\n",
      "    DPVVVV\t204183\n",
      "    SVSVVV\t197597\n",
      "    SVVVVV\t182723\n",
      "    LVVVVV\t158216\n",
      "Time for index table init: 0h 0m 2s 808ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 564446\n",
      "[=================================================================] 2 0s 1ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "11833732 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 333ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp/5123411441862688475/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp/5123411441862688475/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 273ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 1s 153ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp/5123411441862688475/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp/5123411441862688475/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp/5123411441862688475/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/search_tmp/5123411441862688475/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_afdb --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 338ms\n",
      "Time for merging to aln_1rei_afdb: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 875ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/14664210207731514199/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_pdb.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query\n",
      "[=================================================================] 1 0s 6ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 1ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 39ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query_ss databases/PDB/pdb_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp/204530278487467374/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 346125 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 346.12K 0s 108ms\n",
      "Index table: Masked residues: 4289\n",
      "Index table: fill\n",
      "[=================================================================] 346.12K 0s 270ms\n",
      "Index statistics\n",
      "Entries:          62912141\n",
      "DB size:          848 MB\n",
      "Avg k-mer size:   0.983002\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t67034\n",
      "    VVLVVV\t61283\n",
      "    SVSVVV\t57570\n",
      "    VVSVVV\t51410\n",
      "    LVVVVV\t48034\n",
      "    SVVVVV\t47903\n",
      "    DDDVVV\t25399\n",
      "    VVNVVV\t24851\n",
      "    VVQVVV\t22416\n",
      "    VLVVVV\t20962\n",
      "Time for index table init: 0h 0m 1s 141ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 346125\n",
      "[=================================================================] 2 0s 0ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "6473189 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 542ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp/204530278487467374/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp/204530278487467374/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 305ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 1s 243ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp/204530278487467374/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp/204530278487467374/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp/204530278487467374/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/search_tmp/204530278487467374/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_pdb.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 117ms\n",
      "Time for merging to 1rei_pdb.html: 0h 0m 0s 10ms\n",
      "Time for processing: 0h 0m 0s 403ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/13302368596321463924/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query\n",
      "[=================================================================] 1 0s 13ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 2ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query_ss databases/PDB/pdb_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp/12385800545206200641/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 346125 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 346.12K 0s 92ms\n",
      "Index table: Masked residues: 4289\n",
      "Index table: fill\n",
      "[=================================================================] 346.12K 0s 244ms\n",
      "Index statistics\n",
      "Entries:          62912141\n",
      "DB size:          848 MB\n",
      "Avg k-mer size:   0.983002\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t67034\n",
      "    VVLVVV\t61283\n",
      "    SVSVVV\t57570\n",
      "    VVSVVV\t51410\n",
      "    LVVVVV\t48034\n",
      "    SVVVVV\t47903\n",
      "    DDDVVV\t25399\n",
      "    VVNVVV\t24851\n",
      "    VVQVVV\t22416\n",
      "    VLVVVV\t20962\n",
      "Time for index table init: 0h 0m 1s 72ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 346125\n",
      "[=================================================================] 2 0s 1ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "6473189 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 454ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp/12385800545206200641/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp/12385800545206200641/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 377ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 965ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp/12385800545206200641/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp/12385800545206200641/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp/12385800545206200641/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/search_tmp/12385800545206200641/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_pdb --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 333ms\n",
      "Time for merging to aln_1rei_pdb: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 814ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11606409096919011832/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_sp.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query\n",
      "[=================================================================] 1 0s 12ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 43ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query_ss databases/SP/sp_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp/16725278537843415356/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 4G\n",
      "Target database size: 542378 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 542.38K 0s 296ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 542.38K 0s 758ms\n",
      "Index statistics\n",
      "Entries:          165148794\n",
      "DB size:          1433 MB\n",
      "Avg k-mer size:   2.580450\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t206325\n",
      "    SVSVVV\t185871\n",
      "    VVSVVV\t150314\n",
      "    SVVVVV\t138781\n",
      "    VVVSVS\t102688\n",
      "    VSVVVV\t100690\n",
      "    NVSVVV\t94970\n",
      "    VLVLLV\t84161\n",
      "    VSVSVV\t82847\n",
      "    DDVVVV\t82761\n",
      "Time for index table init: 0h 0m 2s 462ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 542378\n",
      "[=================================================================] 2 0s 2ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "11347662 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 7ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp/16725278537843415356/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp/16725278537843415356/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 250ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 1s 565ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp/16725278537843415356/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp/16725278537843415356/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp/16725278537843415356/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/search_tmp/16725278537843415356/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1rei_sp.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 324ms\n",
      "Time for merging to 1rei_sp.html: 0h 0m 0s 22ms\n",
      "Time for processing: 0h 0m 0s 879ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/683391986836139127/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1rei.pdb databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1rei.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query\n",
      "[=================================================================] 1 0s 12ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 2 out of 4.\n",
      "Too short: 2, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 39ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query_ss databases/SP/sp_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp/12481300679221358515/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2 type: Aminoacid\n",
      "Estimated memory consumption: 4G\n",
      "Target database size: 542378 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 542.38K 0s 231ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 542.38K 0s 733ms\n",
      "Index statistics\n",
      "Entries:          165148794\n",
      "DB size:          1433 MB\n",
      "Avg k-mer size:   2.580450\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t206325\n",
      "    SVSVVV\t185871\n",
      "    VVSVVV\t150314\n",
      "    SVVVVV\t138781\n",
      "    VVVSVS\t102688\n",
      "    VSVVVV\t100690\n",
      "    NVSVVV\t94970\n",
      "    VLVLLV\t84161\n",
      "    VSVSVV\t82847\n",
      "    DDVVVV\t82761\n",
      "Time for index table init: 0h 0m 2s 397ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2\n",
      "Target db start 1 to 542378\n",
      "[=================================================================] 2 0s 2ms\n",
      "\n",
      "14050.560748 k-mers per position\n",
      "11347662 DB matches per sequence\n",
      "2 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 2s 899ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp/12481300679221358515/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp/12481300679221358515/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 288ms\n",
      "Time for merging to strualn: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 878ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp/12481300679221358515/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp/12481300679221358515/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp/12481300679221358515/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/search_tmp/12481300679221358515/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1rei_sp --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 2 0s 394ms\n",
      "Time for merging to aln_1rei_sp: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 918ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12416930545057627409/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_up.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query\n",
      "[=================================================================] 1 0s 6ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 1ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 38ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query_ss databases/UP/up_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp/1496118366530664004/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 276G\n",
      "Target database size: 53665860 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 53.67M 19s 57ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 53.67M 1m 0s 421ms\n",
      "Index statistics\n",
      "Entries:          11018018466\n",
      "DB size:          63533 MB\n",
      "Avg k-mer size:   172.156539\n",
      "Top 10 k-mers\n",
      "    DDDDDD\t25475954\n",
      "    DDDDDP\t22448571\n",
      "    DDDDPP\t19428803\n",
      "    VVVVPD\t17790099\n",
      "    DPVVVV\t16769600\n",
      "    SVSVVV\t14134938\n",
      "    SVVVVV\t13707739\n",
      "    DDVVVV\t12394056\n",
      "    LVVVVV\t11880759\n",
      "    VSVVVV\t10654575\n",
      "Time for index table init: 0h 3m 2s 247ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 53665860\n",
      "[=================================================================] 1 0s 1ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "777663266 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 3m 22s 30ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp/1496118366530664004/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp/1496118366530664004/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 66ms\n",
      "Time for merging to strualn: 0h 0m 0s 5ms\n",
      "Time for processing: 0h 1m 15s 267ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp/1496118366530664004/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp/1496118366530664004/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp/1496118366530664004/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/search_tmp/1496118366530664004/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_up.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to 1mbn_up.html: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 10s 171ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/15612761356715825299/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query\n",
      "[=================================================================] 1 0s 7ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query_ss databases/UP/up_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp/18428108341862547155/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 276G\n",
      "Target database size: 53665860 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 53.67M 17s 818ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 53.67M 1m 2s 454ms\n",
      "Index statistics\n",
      "Entries:          11018018466\n",
      "DB size:          63533 MB\n",
      "Avg k-mer size:   172.156539\n",
      "Top 10 k-mers\n",
      "    DDDDDD\t25475954\n",
      "    DDDDDP\t22448571\n",
      "    DDDDPP\t19428803\n",
      "    VVVVPD\t17790099\n",
      "    DPVVVV\t16769600\n",
      "    SVSVVV\t14134938\n",
      "    SVVVVV\t13707739\n",
      "    DDVVVV\t12394056\n",
      "    LVVVVV\t11880759\n",
      "    VSVVVV\t10654575\n",
      "Time for index table init: 0h 3m 2s 388ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 53665860\n",
      "[=================================================================] 1 0s 1ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "777663266 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 3m 21s 828ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp/18428108341862547155/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp/18428108341862547155/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 80ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 1m 11s 564ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp/18428108341862547155/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp/18428108341862547155/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp/18428108341862547155/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/search_tmp/18428108341862547155/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query databases/UP/up FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_up --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 1ms\n",
      "Time for merging to aln_1mbn_up: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 8s 428ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10407519269072328813/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_esma.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query\n",
      "[=================================================================] 1 0s 13ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 1ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query_ss databases/ESMA/esma_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp/18436430245208124192/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 175G\n",
      "Target database size: 36986627 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 36.99M 10s 124ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 36.99M 33s 740ms\n",
      "Index statistics\n",
      "Entries:          6585984425\n",
      "DB size:          38173 MB\n",
      "Avg k-mer size:   102.906007\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t11204510\n",
      "    VVLVVV\t9981536\n",
      "    SVSVVV\t9579775\n",
      "    VVSVVV\t7745515\n",
      "    SVVVVV\t7513922\n",
      "    VSVVVV\t5273955\n",
      "    VVNVVV\t4240901\n",
      "    DPSVVV\t4218901\n",
      "    DDVVVV\t4026947\n",
      "    PPVVVV\t3260480\n",
      "Time for index table init: 0h 1m 34s 616ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 36986627\n",
      "[=================================================================] 1 0s 1ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "534635098 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 1m 45s 880ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp/18436430245208124192/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp/18436430245208124192/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 48ms\n",
      "Time for merging to strualn: 0h 0m 0s 7ms\n",
      "Time for processing: 0h 1m 5s 530ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp/18436430245208124192/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp/18436430245208124192/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp/18436430245208124192/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/search_tmp/18436430245208124192/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_esma.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to 1mbn_esma.html: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 9s 742ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/3118482721551122010/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query\n",
      "[=================================================================] 1 0s 10ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 1ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 42ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query_ss databases/ESMA/esma_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp/13704440299195961584/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 175G\n",
      "Target database size: 36986627 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 36.99M 10s 302ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 36.99M 33s 392ms\n",
      "Index statistics\n",
      "Entries:          6585984425\n",
      "DB size:          38173 MB\n",
      "Avg k-mer size:   102.906007\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t11204510\n",
      "    VVLVVV\t9981536\n",
      "    SVSVVV\t9579775\n",
      "    VVSVVV\t7745515\n",
      "    SVVVVV\t7513922\n",
      "    VSVVVV\t5273955\n",
      "    VVNVVV\t4240901\n",
      "    DPSVVV\t4218901\n",
      "    DDVVVV\t4026947\n",
      "    PPVVVV\t3260480\n",
      "Time for index table init: 0h 1m 33s 794ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 36986627\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "534635098 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 1m 45s 28ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp/13704440299195961584/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp/13704440299195961584/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 69ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 1m 5s 715ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp/13704440299195961584/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp/13704440299195961584/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp/13704440299195961584/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/search_tmp/13704440299195961584/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query databases/ESMA/esma FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_esma --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_1mbn_esma: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 5s 946ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/6334575767891504822/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_afdb.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query\n",
      "[=================================================================] 1 0s 13ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 1ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 43ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query_ss databases/AFDB/afdb_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp/5041471724109755718/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 4G\n",
      "Target database size: 564446 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 564.45K 0s 342ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 564.45K 0s 896ms\n",
      "Index statistics\n",
      "Entries:          178833320\n",
      "DB size:          1511 MB\n",
      "Avg k-mer size:   2.794271\n",
      "Top 10 k-mers\n",
      "    DDDDDD\t343438\n",
      "    DDDDDP\t320181\n",
      "    DDDDPP\t287311\n",
      "    VVVVPD\t239933\n",
      "    VVVVPP\t215141\n",
      "    LVLVVV\t207981\n",
      "    DPVVVV\t204183\n",
      "    SVSVVV\t197597\n",
      "    SVVVVV\t182723\n",
      "    LVVVVV\t158216\n",
      "Time for index table init: 0h 0m 2s 851ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 564446\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "14233995 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 321ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp/5041471724109755718/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp/5041471724109755718/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 92ms\n",
      "Time for merging to strualn: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 1s 689ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp/5041471724109755718/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp/5041471724109755718/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp/5041471724109755718/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/search_tmp/5041471724109755718/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_afdb.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to 1mbn_afdb.html: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 257ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/17048198727530134299/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query\n",
      "[=================================================================] 1 0s 14ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query_ss databases/AFDB/afdb_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp/10786011495296232057/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 4G\n",
      "Target database size: 564446 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 564.45K 0s 304ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 564.45K 0s 953ms\n",
      "Index statistics\n",
      "Entries:          178833320\n",
      "DB size:          1511 MB\n",
      "Avg k-mer size:   2.794271\n",
      "Top 10 k-mers\n",
      "    DDDDDD\t343438\n",
      "    DDDDDP\t320181\n",
      "    DDDDPP\t287311\n",
      "    VVVVPD\t239933\n",
      "    VVVVPP\t215141\n",
      "    LVLVVV\t207981\n",
      "    DPVVVV\t204183\n",
      "    SVSVVV\t197597\n",
      "    SVVVVV\t182723\n",
      "    LVVVVV\t158216\n",
      "Time for index table init: 0h 0m 2s 885ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 564446\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "14233995 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 393ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp/10786011495296232057/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp/10786011495296232057/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 68ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 808ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp/10786011495296232057/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp/10786011495296232057/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp/10786011495296232057/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/search_tmp/10786011495296232057/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query databases/AFDB/afdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_afdb --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 1ms\n",
      "Time for merging to aln_1mbn_afdb: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 411ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/12670794341359370293/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_pdb.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 38ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query_ss databases/PDB/pdb_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp/10759637600417353524/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 346125 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 346.12K 0s 94ms\n",
      "Index table: Masked residues: 4289\n",
      "Index table: fill\n",
      "[=================================================================] 346.12K 0s 242ms\n",
      "Index statistics\n",
      "Entries:          62912141\n",
      "DB size:          848 MB\n",
      "Avg k-mer size:   0.983002\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t67034\n",
      "    VVLVVV\t61283\n",
      "    SVSVVV\t57570\n",
      "    VVSVVV\t51410\n",
      "    LVVVVV\t48034\n",
      "    SVVVVV\t47903\n",
      "    DDDVVV\t25399\n",
      "    VVNVVV\t24851\n",
      "    VVQVVV\t22416\n",
      "    VLVVVV\t20962\n",
      "Time for index table init: 0h 0m 1s 67ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 346125\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "3758580 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 390ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp/10759637600417353524/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp/10759637600417353524/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 102ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 1s 171ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp/10759637600417353524/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp/10759637600417353524/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp/10759637600417353524/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/search_tmp/10759637600417353524/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_pdb.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to 1mbn_pdb.html: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 237ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/4325937093111978710/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query\n",
      "[=================================================================] 1 0s 13ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 39ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query_ss databases/PDB/pdb_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp/552469308235722629/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 346125 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 346.12K 0s 100ms\n",
      "Index table: Masked residues: 4289\n",
      "Index table: fill\n",
      "[=================================================================] 346.12K 0s 270ms\n",
      "Index statistics\n",
      "Entries:          62912141\n",
      "DB size:          848 MB\n",
      "Avg k-mer size:   0.983002\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t67034\n",
      "    VVLVVV\t61283\n",
      "    SVSVVV\t57570\n",
      "    VVSVVV\t51410\n",
      "    LVVVVV\t48034\n",
      "    SVVVVV\t47903\n",
      "    DDDVVV\t25399\n",
      "    VVNVVV\t24851\n",
      "    VVQVVV\t22416\n",
      "    VLVVVV\t20962\n",
      "Time for index table init: 0h 0m 1s 100ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 346125\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "3758580 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 431ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp/552469308235722629/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp/552469308235722629/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 127ms\n",
      "Time for merging to strualn: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 867ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp/552469308235722629/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp/552469308235722629/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp/552469308235722629/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/search_tmp/552469308235722629/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query databases/PDB/pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_pdb --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 5ms\n",
      "Time for merging to aln_1mbn_pdb: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 716ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/10437500074712698138/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_sp.html FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-mode 3 \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t3\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query\n",
      "[=================================================================] 1 0s 12ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 39ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query_ss databases/SP/sp_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp/6213359254313043071/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 4G\n",
      "Target database size: 542378 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 542.38K 0s 261ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 542.38K 0s 717ms\n",
      "Index statistics\n",
      "Entries:          165148794\n",
      "DB size:          1433 MB\n",
      "Avg k-mer size:   2.580450\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t206325\n",
      "    SVSVVV\t185871\n",
      "    VVSVVV\t150314\n",
      "    SVVVVV\t138781\n",
      "    VVVSVS\t102688\n",
      "    VSVVVV\t100690\n",
      "    NVSVVV\t94970\n",
      "    VLVLLV\t84161\n",
      "    VSVSVV\t82847\n",
      "    DDVVVV\t82761\n",
      "Time for index table init: 0h 0m 2s 376ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 542378\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "12966862 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 2s 842ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp/6213359254313043071/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp/6213359254313043071/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 65ms\n",
      "Time for merging to strualn: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 1s 481ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp/6213359254313043071/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp/6213359254313043071/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp/6213359254313043071/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/search_tmp/6213359254313043071/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/result FoldSeekExamples/FoldSeekExtractionOutputHTMLs/1mbn_sp.html --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 3 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to 1mbn_sp.html: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 276ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/7892426013660651541/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp\n",
      "easy-search FoldSeekExamples/ExampleQueries/1mbn.pdb databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage \n",
      "\n",
      "MMseqs Version:              \t5433d6db83522482410b30b180001eeb269f7505\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t72\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "Alignment backtraces will be computed, since they were requested by output format.\n",
      "createdb FoldSeekExamples/ExampleQueries/1mbn.pdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 72 -v 3 \n",
      "\n",
      "Output file: FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query\n",
      "[=================================================================] 1 0s 14ms\n",
      "Time for merging to query_ss: 0h 0m 0s 2ms\n",
      "Time for merging to query_h: 0h 0m 0s 2ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 1ms\n",
      "Ignore 0 out of 1.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 41ms\n",
      "Create directory FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp\n",
      "search FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp -a 1 --alignment-mode 3 --comp-bias-corr 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 -s 9.5 -k 6 --mask 0 --mask-prob 0.99995 --remove-tmp-files 1 \n",
      "\n",
      "prefilter FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query_ss databases/SP/sp_ss FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp/17629711614652906159/pref --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 9.5 -k 6 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.99995 --mask-lower-case 1 --min-ungapped-score 30 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 4G\n",
      "Target database size: 542378 type: Aminoacid\n",
      "Index table k-mer threshold: 78 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 542.38K 0s 260ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 542.38K 0s 767ms\n",
      "Index statistics\n",
      "Entries:          165148794\n",
      "DB size:          1433 MB\n",
      "Avg k-mer size:   2.580450\n",
      "Top 10 k-mers\n",
      "    LVLVVV\t206325\n",
      "    SVSVVV\t185871\n",
      "    VVSVVV\t150314\n",
      "    SVVVVV\t138781\n",
      "    VVVSVS\t102688\n",
      "    VSVVVV\t100690\n",
      "    NVSVVV\t94970\n",
      "    VLVLLV\t84161\n",
      "    VSVSVV\t82847\n",
      "    DDVVVV\t82761\n",
      "Time for index table init: 0h 0m 2s 361ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 78\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 542378\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "162.503268 k-mers per position\n",
      "12966862 DB matches per sequence\n",
      "1 overflows\n",
      "1000 sequences passed prefiltering per query sequence\n",
      "1000 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 2s 797ms\n",
      "structurealign FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp/17629711614652906159/pref FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp/17629711614652906159/strualn --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 1 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 1 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 10 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 0.5 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 48ms\n",
      "Time for merging to strualn: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 818ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp/17629711614652906159/strualn FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp/17629711614652906159/aln \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp/17629711614652906159/aln FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Removing temporary files\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/search_tmp/17629711614652906159/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "convertalis FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query databases/SP/sp FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/result FoldSeekExamples/FoldSeekExtractionOutputAlntmps/aln_1mbn_sp --sub-mat 'aa:3di.out,nucl:3di.out' --format-mode 0 --format-output query,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,tseq,evalue,bits,cigar,alntmscore,qtmscore,ttmscore,rmsd,u,t,lddt,lddtfull,prob,taxid,taxname,taxlineage --translation-table 1 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --db-output 0 --db-load-mode 0 --search-type 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 1ms\n",
      "Time for merging to aln_1mbn_sp: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 757ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/result -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query_ca -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb FoldSeekExamples/FoldSeekExtractionOutputAlntmps/tmp/11978640814391548399/query_ss -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n"
     ]
    }
   ],
   "source": [
    "FirstFoldSeekdf = pd.DataFrame(columns=Columns) # This is the first major dataframe, it will be the largest, and saved at the end of the cell\n",
    "# Loop through all the Query proteins\n",
    "for QueryProtein in QueryProteins:\n",
    "    PDBidentifier = QueryProtein.split('/')[-1].split('.')[-2] # This should hold true for any path to any .pdb file, this variable is used for naming of files\n",
    "\n",
    "    for Database in Databases:     # Loop through all the databases\n",
    "        \n",
    "        #First make the HTMLS (This doesn't depend on if the database is ESMA or not)\n",
    "        if CreateHTMLs:\n",
    "                os.system(f\"foldseek easy-search {QueryProtein} {Database} {HTMLFileDirectory}/{PDBidentifier}_{Database.split('/')[-1]}.html {AlignmentAndTempFileDirectory}/tmp --format-mode 3\")\n",
    "                os.system(f\"rm -r {AlignmentAndTempFileDirectory}/tmp\") # remove the temporary file (for storage and sanity)\n",
    "            \n",
    "        # Generate the alignment file - with a different output format if the database is ESMA    \n",
    "        if Database.split('/')[-1] == 'esma':\n",
    "            os.system(f\"foldseek easy-search {QueryProtein} {Database} {AlignmentAndTempFileDirectory}/aln_{PDBidentifier}_{Database.split('/')[-1]} {AlignmentAndTempFileDirectory}/tmp {FormatOutputESMA}\")\n",
    "            df = pd.read_csv(f\"{AlignmentAndTempFileDirectory}/aln_{PDBidentifier}_{Database.split('/')[-1]}\",sep='\\t', header=None) # Read the current alignment file (it is a tab separated file)\n",
    "            df.columns = Columns[:-4]\n",
    "        else:\n",
    "            os.system(f\"foldseek easy-search {QueryProtein} {Database} {AlignmentAndTempFileDirectory}/aln_{PDBidentifier}_{Database.split('/')[-1]} {AlignmentAndTempFileDirectory}/tmp {FormatOutput}\")\n",
    "            df = pd.read_csv(f\"{AlignmentAndTempFileDirectory}/aln_{PDBidentifier}_{Database.split('/')[-1]}\",sep='\\t', header=None) # Read the current alignment file (it is a tab separated file)\n",
    "            df.columns = Columns[:-1]\n",
    "        \n",
    "        df['database'] = Database # Add in what database the current alignments are from\n",
    "        df = df.reindex(columns=Columns) # Just to make sure everything is as it should be - the ESMA df should have three NaN columns\n",
    "        \n",
    "        os.system(f\"rm -r {AlignmentAndTempFileDirectory}/tmp\") # remove the temporary directory (for storage and sanity)\n",
    "        FirstFoldSeekdf = pd.concat([df,FirstFoldSeekdf],axis=0) # Add the current alignment to the main dataframe\n",
    "\n",
    "FirstFoldSeekdf.to_csv(f'{CSVFileDirectory}/FirstFoldSeekdf.csv',index=False) # Save this dataframe to a csv for future manipulation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03f70d57-7b76-4eef-9fe8-64e94b576d2a",
   "metadata": {},
   "source": [
    "### This may be a good point to make your own notebook and investigate the data held in the newly created csv yourself"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "09a8e2b3-f2c4-470c-ae90-0639f75220bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "FirstFoldSeekdf = pd.read_csv(f'{CSVFileDirectory}/FirstFoldSeekdf.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "4ef9c2d2-2073-42f1-83ea-8f7366dfe354",
   "metadata": {},
   "outputs": [],
   "source": [
    "FirstFoldSeekdf['QueryProteinLength'] = 0 # Lets initialise these columns and fill them in, this data may be useful downstream\n",
    "FirstFoldSeekdf['TargetProteinLength'] = FirstFoldSeekdf['tseq'].apply(lambda x: len(x))\n",
    "\n",
    "for query in list(FirstFoldSeekdf['query'].unique()): # we are looping through each unique query to calculate their length\n",
    "    queryUnique = query.split('_')[0] # pdb files can contain multiple chains\n",
    "    NumChains = sum(query.split('_')[0] in pdbfile for pdbfile in list(FirstFoldSeekdf['query'].unique())) # count chains - need to divide by this after as it will count amino acids from all chains!\n",
    "    ProteinLength = calculate_protein_length(f'{QueryPath}/{queryUnique}')/NumChains\n",
    "    FirstFoldSeekdf.loc[FirstFoldSeekdf['query'] == query, 'QueryProteinLength'] = ProteinLength"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "995614f7-2da5-4c31-bb21-0218707ca7bf",
   "metadata": {},
   "source": [
    "## Obtaining variables from BLASTs\n",
    "### NOTE: There will be many repeated target sequences - infact *most of them will be* if you have many structurally similar queries - to not make the BLASTs horrifically inefficent, we will take the unique sequences and BLAST those and then join back to the main dataframe using the sequences('tseq') column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "8b65ed90-9c49-4459-adcf-92203b6d8035",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MVLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MVLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MVLSEGEWQLVLHVWAKVEADIAGHGQDILIRLFKHHPETLEKFDR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MVLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MGLSDGEWQLVLNVWGKVEADLAGHGQDILIRLFKGHPETLEKFDK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8725</th>\n",
       "      <td>VCFAMLLLVLPLGRKRNGQVTVTQPPASVRLEGTVTISATGSSDIG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8726</th>\n",
       "      <td>TTNLLNDTKCTGFLLWLIHTDKSKSVSLGATVTISATGSSNIGADL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8727</th>\n",
       "      <td>MSCCYSYILDEKQTELDFVICFCTASQGVDDDLSCYLQKPGQPPKL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8728</th>\n",
       "      <td>MTPTADSCPSSLLTQSPSLAGPVRVLAVSWYQQQQRKAPKLLIYYA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8729</th>\n",
       "      <td>MTFGIPLWMSTDLTATLVLTLFPPVHSVQPGDTMQILYRASQDIGN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8730 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   tseq\n",
       "0     MVLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDR...\n",
       "1     MVLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDK...\n",
       "2     MVLSEGEWQLVLHVWAKVEADIAGHGQDILIRLFKHHPETLEKFDR...\n",
       "3     MVLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDK...\n",
       "4     MGLSDGEWQLVLNVWGKVEADLAGHGQDILIRLFKGHPETLEKFDK...\n",
       "...                                                 ...\n",
       "8725  VCFAMLLLVLPLGRKRNGQVTVTQPPASVRLEGTVTISATGSSDIG...\n",
       "8726  TTNLLNDTKCTGFLLWLIHTDKSKSVSLGATVTISATGSSNIGADL...\n",
       "8727  MSCCYSYILDEKQTELDFVICFCTASQGVDDDLSCYLQKPGQPPKL...\n",
       "8728  MTPTADSCPSSLLTQSPSLAGPVRVLAVSWYQQQQRKAPKLLIYYA...\n",
       "8729  MTFGIPLWMSTDLTATLVLTLFPPVHSVQPGDTMQILYRASQDIGN...\n",
       "\n",
       "[8730 rows x 1 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "UniqueBLASTDF = pd.DataFrame()\n",
    "UniqueBLASTDF['tseq'] = FirstFoldSeekdf['tseq'].unique()\n",
    "UniqueBLASTDF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4ed651c2-0f0d-4f66-8087-46f015b0eac6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This can take just as much time as the FoldSeek itself depending on how many BLASTs you're doing and the database size!\n",
    "\n",
    "for BLAST in BLASTList: # Loop through each BLAST we need to do and BLAST the whole 'tseq' column\n",
    "    BLASTInput = glob.glob(f'BLASTDatabases/{BLAST}_BLAST_DB/*.pdb')[0].split('.pdb')[0] # this will be the .fa file for those rebuilt, or for those not rebuilt\n",
    "                                                                 # from fasta e.g. patent - just the string 'pataa' which is what BLAST wants\n",
    "    BLASTOutput = f'{BLASTOutputDirectory}/{BLAST}_Output.xml' # Specific output .xml file for this BLAST\n",
    "    GetBestBlastAlignmentBitscoreEff(UniqueBLASTDF['tseq'], # This function BLASTs the whole column at once\n",
    "                                  BLASTInput,\n",
    "                                  BLASTOutputDirectory,\n",
    "                                  BLASTOutput)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "562c6c67-5bb2-4506-acd2-35fb9a6f7d5d",
   "metadata": {},
   "source": [
    "### I split up these loops as running BLAST requires significantly more computation than parsing, and conceptually it's helpful to split them up "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "be94e86b-0814-428b-b5ca-dccb75172d28",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tseq</th>\n",
       "      <th>BPPRC Blastbitscores</th>\n",
       "      <th>BPPRC Blast raw_bistscores</th>\n",
       "      <th>BPPRC Blast eval</th>\n",
       "      <th>BPPRC Blast aln len</th>\n",
       "      <th>BPPRC Blast name</th>\n",
       "      <th>BPPRC Blast positives</th>\n",
       "      <th>BPPRC Blast identities</th>\n",
       "      <th>BPPRC Blast gaps</th>\n",
       "      <th>AllergenOnline Blastbitscores</th>\n",
       "      <th>...</th>\n",
       "      <th>Patent Blast raw_bistscores</th>\n",
       "      <th>Patent Blast eval</th>\n",
       "      <th>Patent Blast aln len</th>\n",
       "      <th>Patent Blast name</th>\n",
       "      <th>Patent Blast positives</th>\n",
       "      <th>Patent Blast identities</th>\n",
       "      <th>Patent Blast gaps</th>\n",
       "      <th>BPPRC Blast bitscores</th>\n",
       "      <th>AllergenOnline Blast bitscores</th>\n",
       "      <th>Patent Blast bitscores</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MVLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDR...</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0</td>\n",
       "      <td>10.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>no protein</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35.4242</td>\n",
       "      <td>...</td>\n",
       "      <td>791</td>\n",
       "      <td>2.472320e-108</td>\n",
       "      <td>154</td>\n",
       "      <td>Sequence 10 from patent US 10934531</td>\n",
       "      <td>154</td>\n",
       "      <td>154</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>35.4242</td>\n",
       "      <td>309.3010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MVLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDK...</td>\n",
       "      <td>25.4090</td>\n",
       "      <td>54</td>\n",
       "      <td>1.329000</td>\n",
       "      <td>95</td>\n",
       "      <td></td>\n",
       "      <td>39</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>35.0390</td>\n",
       "      <td>...</td>\n",
       "      <td>785</td>\n",
       "      <td>1.594730e-107</td>\n",
       "      <td>153</td>\n",
       "      <td>Sequence 2 from patent US 7164991</td>\n",
       "      <td>153</td>\n",
       "      <td>153</td>\n",
       "      <td>0</td>\n",
       "      <td>25.4090</td>\n",
       "      <td>35.0390</td>\n",
       "      <td>306.9900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MVLSEGEWQLVLHVWAKVEADIAGHGQDILIRLFKHHPETLEKFDR...</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0</td>\n",
       "      <td>10.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>no protein</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35.4242</td>\n",
       "      <td>...</td>\n",
       "      <td>768</td>\n",
       "      <td>6.753590e-105</td>\n",
       "      <td>154</td>\n",
       "      <td>Sequence 10 from patent US 10934531</td>\n",
       "      <td>152</td>\n",
       "      <td>149</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>35.4242</td>\n",
       "      <td>300.4420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MVLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDK...</td>\n",
       "      <td>25.4090</td>\n",
       "      <td>54</td>\n",
       "      <td>1.329000</td>\n",
       "      <td>95</td>\n",
       "      <td></td>\n",
       "      <td>39</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>35.0390</td>\n",
       "      <td>...</td>\n",
       "      <td>780</td>\n",
       "      <td>1.149810e-106</td>\n",
       "      <td>153</td>\n",
       "      <td>Sequence 2 from patent US 7164991</td>\n",
       "      <td>152</td>\n",
       "      <td>152</td>\n",
       "      <td>0</td>\n",
       "      <td>25.4090</td>\n",
       "      <td>35.0390</td>\n",
       "      <td>305.0640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MGLSDGEWQLVLNVWGKVEADLAGHGQDILIRLFKGHPETLEKFDK...</td>\n",
       "      <td>27.3350</td>\n",
       "      <td>59</td>\n",
       "      <td>0.342556</td>\n",
       "      <td>59</td>\n",
       "      <td></td>\n",
       "      <td>31</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>33.1130</td>\n",
       "      <td>...</td>\n",
       "      <td>729</td>\n",
       "      <td>7.370430e-99</td>\n",
       "      <td>154</td>\n",
       "      <td>Sequence 10 from patent US 10934531</td>\n",
       "      <td>149</td>\n",
       "      <td>140</td>\n",
       "      <td>0</td>\n",
       "      <td>27.3350</td>\n",
       "      <td>33.1130</td>\n",
       "      <td>285.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8725</th>\n",
       "      <td>VCFAMLLLVLPLGRKRNGQVTVTQPPASVRLEGTVTISATGSSDIG...</td>\n",
       "      <td>24.2534</td>\n",
       "      <td>51</td>\n",
       "      <td>3.031490</td>\n",
       "      <td>46</td>\n",
       "      <td></td>\n",
       "      <td>21</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>25.0238</td>\n",
       "      <td>...</td>\n",
       "      <td>285</td>\n",
       "      <td>2.538960e-32</td>\n",
       "      <td>99</td>\n",
       "      <td>Sequence 5 from patent US 9758586</td>\n",
       "      <td>74</td>\n",
       "      <td>59</td>\n",
       "      <td>4</td>\n",
       "      <td>24.2534</td>\n",
       "      <td>25.0238</td>\n",
       "      <td>114.3900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8726</th>\n",
       "      <td>TTNLLNDTKCTGFLLWLIHTDKSKSVSLGATVTISATGSSNIGADL...</td>\n",
       "      <td>23.0978</td>\n",
       "      <td>48</td>\n",
       "      <td>4.796600</td>\n",
       "      <td>37</td>\n",
       "      <td></td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>22.7126</td>\n",
       "      <td>...</td>\n",
       "      <td>293</td>\n",
       "      <td>8.385620e-34</td>\n",
       "      <td>93</td>\n",
       "      <td>Sequence 12 from patent US 11008400</td>\n",
       "      <td>71</td>\n",
       "      <td>57</td>\n",
       "      <td>0</td>\n",
       "      <td>23.0978</td>\n",
       "      <td>22.7126</td>\n",
       "      <td>117.4720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8727</th>\n",
       "      <td>MSCCYSYILDEKQTELDFVICFCTASQGVDDDLSCYLQKPGQPPKL...</td>\n",
       "      <td>25.0238</td>\n",
       "      <td>53</td>\n",
       "      <td>0.979482</td>\n",
       "      <td>59</td>\n",
       "      <td></td>\n",
       "      <td>25</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>24.2534</td>\n",
       "      <td>...</td>\n",
       "      <td>222</td>\n",
       "      <td>5.478150e-23</td>\n",
       "      <td>74</td>\n",
       "      <td>Sequence 8 from patent US 10266600</td>\n",
       "      <td>51</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>25.0238</td>\n",
       "      <td>24.2534</td>\n",
       "      <td>90.1225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8728</th>\n",
       "      <td>MTPTADSCPSSLLTQSPSLAGPVRVLAVSWYQQQQRKAPKLLIYYA...</td>\n",
       "      <td>25.7942</td>\n",
       "      <td>55</td>\n",
       "      <td>0.413379</td>\n",
       "      <td>39</td>\n",
       "      <td></td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>25.4090</td>\n",
       "      <td>...</td>\n",
       "      <td>270</td>\n",
       "      <td>2.325440e-30</td>\n",
       "      <td>79</td>\n",
       "      <td>Sequence 17034 from patent US 9254311</td>\n",
       "      <td>59</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>25.7942</td>\n",
       "      <td>25.4090</td>\n",
       "      <td>108.6120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8729</th>\n",
       "      <td>MTFGIPLWMSTDLTATLVLTLFPPVHSVQPGDTMQILYRASQDIGN...</td>\n",
       "      <td>26.1794</td>\n",
       "      <td>56</td>\n",
       "      <td>1.258910</td>\n",
       "      <td>42</td>\n",
       "      <td></td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>26.5646</td>\n",
       "      <td>...</td>\n",
       "      <td>314</td>\n",
       "      <td>3.304100e-34</td>\n",
       "      <td>204</td>\n",
       "      <td>Sequence 257 from patent US 9464131</td>\n",
       "      <td>111</td>\n",
       "      <td>76</td>\n",
       "      <td>18</td>\n",
       "      <td>26.1794</td>\n",
       "      <td>26.5646</td>\n",
       "      <td>125.5610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8730 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   tseq  BPPRC Blastbitscores  \\\n",
       "0     MVLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDR...                0.0000   \n",
       "1     MVLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDK...               25.4090   \n",
       "2     MVLSEGEWQLVLHVWAKVEADIAGHGQDILIRLFKHHPETLEKFDR...                0.0000   \n",
       "3     MVLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDK...               25.4090   \n",
       "4     MGLSDGEWQLVLNVWGKVEADLAGHGQDILIRLFKGHPETLEKFDK...               27.3350   \n",
       "...                                                 ...                   ...   \n",
       "8725  VCFAMLLLVLPLGRKRNGQVTVTQPPASVRLEGTVTISATGSSDIG...               24.2534   \n",
       "8726  TTNLLNDTKCTGFLLWLIHTDKSKSVSLGATVTISATGSSNIGADL...               23.0978   \n",
       "8727  MSCCYSYILDEKQTELDFVICFCTASQGVDDDLSCYLQKPGQPPKL...               25.0238   \n",
       "8728  MTPTADSCPSSLLTQSPSLAGPVRVLAVSWYQQQQRKAPKLLIYYA...               25.7942   \n",
       "8729  MTFGIPLWMSTDLTATLVLTLFPPVHSVQPGDTMQILYRASQDIGN...               26.1794   \n",
       "\n",
       "      BPPRC Blast raw_bistscores  BPPRC Blast eval  BPPRC Blast aln len  \\\n",
       "0                              0         10.000000                    0   \n",
       "1                             54          1.329000                   95   \n",
       "2                              0         10.000000                    0   \n",
       "3                             54          1.329000                   95   \n",
       "4                             59          0.342556                   59   \n",
       "...                          ...               ...                  ...   \n",
       "8725                          51          3.031490                   46   \n",
       "8726                          48          4.796600                   37   \n",
       "8727                          53          0.979482                   59   \n",
       "8728                          55          0.413379                   39   \n",
       "8729                          56          1.258910                   42   \n",
       "\n",
       "     BPPRC Blast name  BPPRC Blast positives  BPPRC Blast identities  \\\n",
       "0          no protein                      0                       0   \n",
       "1                                         39                      23   \n",
       "2          no protein                      0                       0   \n",
       "3                                         39                      23   \n",
       "4                                         31                      17   \n",
       "...               ...                    ...                     ...   \n",
       "8725                                      21                      16   \n",
       "8726                                      21                      13   \n",
       "8727                                      25                      17   \n",
       "8728                                      22                      11   \n",
       "8729                                      22                      13   \n",
       "\n",
       "      BPPRC Blast gaps  AllergenOnline Blastbitscores  ...  \\\n",
       "0                    0                        35.4242  ...   \n",
       "1                    4                        35.0390  ...   \n",
       "2                    0                        35.4242  ...   \n",
       "3                    4                        35.0390  ...   \n",
       "4                    4                        33.1130  ...   \n",
       "...                ...                            ...  ...   \n",
       "8725                 8                        25.0238  ...   \n",
       "8726                 0                        22.7126  ...   \n",
       "8727                 2                        24.2534  ...   \n",
       "8728                 2                        25.4090  ...   \n",
       "8729                 3                        26.5646  ...   \n",
       "\n",
       "      Patent Blast raw_bistscores  Patent Blast eval  Patent Blast aln len  \\\n",
       "0                             791      2.472320e-108                   154   \n",
       "1                             785      1.594730e-107                   153   \n",
       "2                             768      6.753590e-105                   154   \n",
       "3                             780      1.149810e-106                   153   \n",
       "4                             729       7.370430e-99                   154   \n",
       "...                           ...                ...                   ...   \n",
       "8725                          285       2.538960e-32                    99   \n",
       "8726                          293       8.385620e-34                    93   \n",
       "8727                          222       5.478150e-23                    74   \n",
       "8728                          270       2.325440e-30                    79   \n",
       "8729                          314       3.304100e-34                   204   \n",
       "\n",
       "                          Patent Blast name  Patent Blast positives  \\\n",
       "0       Sequence 10 from patent US 10934531                     154   \n",
       "1         Sequence 2 from patent US 7164991                     153   \n",
       "2       Sequence 10 from patent US 10934531                     152   \n",
       "3         Sequence 2 from patent US 7164991                     152   \n",
       "4       Sequence 10 from patent US 10934531                     149   \n",
       "...                                     ...                     ...   \n",
       "8725      Sequence 5 from patent US 9758586                      74   \n",
       "8726    Sequence 12 from patent US 11008400                      71   \n",
       "8727     Sequence 8 from patent US 10266600                      51   \n",
       "8728  Sequence 17034 from patent US 9254311                      59   \n",
       "8729    Sequence 257 from patent US 9464131                     111   \n",
       "\n",
       "      Patent Blast identities  Patent Blast gaps  BPPRC Blast bitscores  \\\n",
       "0                         154                  0                 0.0000   \n",
       "1                         153                  0                25.4090   \n",
       "2                         149                  0                 0.0000   \n",
       "3                         152                  0                25.4090   \n",
       "4                         140                  0                27.3350   \n",
       "...                       ...                ...                    ...   \n",
       "8725                       59                  4                24.2534   \n",
       "8726                       57                  0                23.0978   \n",
       "8727                       43                  0                25.0238   \n",
       "8728                       54                  0                25.7942   \n",
       "8729                       76                 18                26.1794   \n",
       "\n",
       "      AllergenOnline Blast bitscores  Patent Blast bitscores  \n",
       "0                            35.4242                309.3010  \n",
       "1                            35.0390                306.9900  \n",
       "2                            35.4242                300.4420  \n",
       "3                            35.0390                305.0640  \n",
       "4                            33.1130                285.4190  \n",
       "...                              ...                     ...  \n",
       "8725                         25.0238                114.3900  \n",
       "8726                         22.7126                117.4720  \n",
       "8727                         24.2534                 90.1225  \n",
       "8728                         25.4090                108.6120  \n",
       "8729                         26.5646                125.5610  \n",
       "\n",
       "[8730 rows x 28 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for BLAST in BLASTList:\n",
    "    UniqueBLASTDF = add_blast(UniqueBLASTDF,f'{BLASTOutputDirectory}/{BLAST}_Output.xml',BLAST)\n",
    "UniqueBLASTDF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "3b8d34b7-37f6-40c4-b5f1-4181b94b8f8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "BLASTeddf = pd.merge(FirstFoldSeekdf, UniqueBLASTDF, on='tseq', how='inner')\n",
    "BLASTeddf.to_csv(f'{CSVFileDirectory}/BLASTedFoldSeekdf.csv',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2b3b913-54c5-4497-a4d3-d8cc6d368922",
   "metadata": {},
   "source": [
    "### Here may again be a good place to stop and visualise what all this extra data looks like"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3b02035-9fe6-4f4d-ba2d-2b8e9d0f7668",
   "metadata": {},
   "source": [
    "## Filtering based on structure and BLAST\n",
    "### This stage is entirely User-dependent, filter don't filter, noodle don't noodle... It's all up to you (and those parameters you set at the start)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "a04950ee-2575-42a8-8437-211055abd6c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "StructFiltdf = BLASTeddf[(BLASTeddf['qtmscore']>=TMScoreThreshold) & (BLASTeddf['ttmscore']>=TMScoreThreshold)]\n",
    "StructFiltdf.to_csv(f'{CSVFileDirectory}/StructFiltdf.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "4ad0de24-2e7a-4f9f-8fbd-78726dc2fdc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "BLASTFiltdf = StructFiltdf\n",
    "for BLAST, thresh in zip(BLASTList, BLASTBitscoreThresholds): # Make sure the lists are the same length!\n",
    "    BLASTFiltdf = BLASTFiltdf[BLASTFiltdf[f'{BLAST} Blast bitscores'] <= thresh]\n",
    "BLASTFiltdf.to_csv(f'{CSVFileDirectory}/BLASTFiltdf.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "3cd81ede-323e-4d4e-8d82-d4828c7204c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "if MustStartWithMethionine:\n",
    "    MethionineFiltdf = BLASTFiltdf[BLASTFiltdf['tseq'].str.startswith('M')]\n",
    "    MethionineFiltdf.to_csv(f'{CSVFileDirectory}/MethionineFiltdf.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "6b97c6de-dd8c-4c66-956b-38e0787b4d5e",
   "metadata": {},
   "outputs": [],
   "source": [
    "if RemoveDuplicates:\n",
    "    DuplicateRemoveddf = MethionineFiltdf.drop_duplicates(subset='tseq')\n",
    "    DuplicateRemoveddf.to_csv(f'{CSVFileDirectory}/DuplicateRemoveddf.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfe315ec-ab02-4349-bde7-e1f6cdc06385",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
